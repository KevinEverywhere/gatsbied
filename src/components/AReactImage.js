import React from 'react'
import PropTypes from 'prop-types'

const AReactImage = (props) => {
  // console.log(props);
  const radius = {
    primitive: 'sphere',
    radius:props.radius
  }
  if(props.spherical){
    return(
    <a-entity geometry={radius} position={props.position} rotation={props.rotation}>
    <a-image geometry={radius} src={props.src} position="0 0 0" />
    </a-entity>
  )
  }else{
    return(<a-entity position={props.position} rotation={props.rotation}>
      <a-image material="opacity:0.8" src={props.src} width={props.width} height={props.height} position="0 0 0" />
    </a-entity>)
  }
}

AReactImage.propTypes={
  position: PropTypes.string,
  rotation: PropTypes.string,
  radius: PropTypes.string,
  src: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
}

export default AReactImage

/**
 * <AReactImage src={item} rotation={rot} width={distanceBetween} height={distanceBetween} position={pos} />
 */
