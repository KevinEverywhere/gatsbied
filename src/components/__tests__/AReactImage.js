import React from "react"
import renderer from "react-test-renderer"
import AReactImage from "../AReactImage"

describe("AReactImage", () => {
  it("renders correctly", () => {
    const fluid = {src:''}
    const pos = '0 0 0';
    const distanceBetween = 10;
    const tree = renderer
      .create(<AReactImage fluid={fluid} rotation={pos} width={distanceBetween} height={distanceBetween} position={pos} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
