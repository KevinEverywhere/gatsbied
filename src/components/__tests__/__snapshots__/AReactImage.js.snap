// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AReactImage renders correctly 1`] = `
<a-entity
  position="0 0 0"
  rotation="0 0 0"
>
  <a-image
    height={10}
    material="opacity:0.8"
    position="0 0 0"
    src=""
    width={10}
  />
</a-entity>
`;
